projects/igniteui-angular/src/lib/date-range-picker/date-range-picker.component.ts

```diff
@@ -569,12 +569,27 @@ export class IgxDateRangePickerComponent extends DisplayDensityBase
 
     /** @hidden @internal */
     public validate(control: AbstractControl): ValidationErrors {
-        if (this.minValue && !this.valueInRange(control.value)) {
+        const value: DateRange = control.value;
+        if (this.minValue && value && value.start && DatePickerUtil.lessThanMinValue(value.start, this.minValue, false)) {
+            return { 'minValue': true };
+        }
+        if (this.minValue && value && value.end && DatePickerUtil.lessThanMinValue(value.end, this.minValue, false)) {
             return { 'minValue': true };
         }
-        if (this.maxValue && !this.valueInRange(control.value)) {
+        if (this.maxValue && value && value.start && DatePickerUtil.greaterThanMaxValue(value.start, this.maxValue, false)) {
+            return { 'maxValue': true };
+        }
+        if (this.maxValue && value && value.end && DatePickerUtil.greaterThanMaxValue(value.end, this.maxValue, false)) {
             return { 'maxValue': true };
         }
+        if (control.validator) {
```

Comment:
This should definitely not be needed. As `Validator` you are part of the form control validators that are all called anyway. You only provide your bit of the check and for sure don't call the entire validation again (that should've even entered an endless loop with the call below).

Reply:
when we have required on `ngModel` angular test if component's `value` is empty like this:
```
function isEmptyInputValue(value) {
    // we don't check for string here so it also works with arrays
    return value == null || value.length === 0;
}
```
If our `value` is `{ start: null, end: null }` angular is Ok with this `value` and does not set `required` to `true`.

And no this will not enter infinite loop as we are calling `validator` against empty object

Reply:
when we have required on `ngModel` angular test if component's `value` is empty like this:
```
function isEmptyInputValue(value) {
    // we don't check for string here so it also works with arrays
    return value == null || value.length === 0;
}
```
If our `value` is `{ start: null, end: null }` angular is Ok with this `value` and does not set `required` to `true`.

And no this will not enter infinite loop as we are calling `validator` against empty object

Reply:
Fairly certain you are calling validator function (which combines all validator functions, including this one) with an empty control (normally called with `this`), that will merely have `undefined` value. If the required validator runs, so should this one, so still a chance this will loop.

In either case - the required check is fine, there's very little reason to have an object with two null values. If there's no range and the value is cleared I'd just have it as `null`. Plus it's easier to check if `value` exist rather than going `value.start && value.end`

Reply:
what if we have value and start or end is null - it is still not valid. And yes we do hope one will set range to null if both start and end are null, but will they all do it?

Reply:
First - the scenario is a bit odd, since that's only possible with projected inputs that are meant to validate on their own (i.e. have their own models). And ideally the control itself should not enter this state at all (the TODO you've left). At the end of the day if a user forcibly assigns a range with nulls, it should be acceptable to also add a custom validator to handle it.